### PR TITLE
unit test torchaudio rnnt loss

### PR DESCRIPTION
**Pull request by carolineechen**
Currently running into the following error, where the computed loss w/ torchaudio is twice the expected loss

```
E       assert 4.495666980743408 == 2.247833251953125
E        +  where 4.495666980743408 = <built-in method item of Tensor object at 0x7f7cc07cdbd0>()
E        +    where <built-in method item of Tensor object at 0x7f7cc07cdbd0> = tensor(4.4957, grad_fn=<MeanBackward0>).item
```

### CODE REVIEW

File: tests/unittests/test_losses.py [hunk 1]
```diff
@@ -137,3 +138,43 @@ def test_transducer_loss():
...
+                        [0.7, 0.1, 0.2, 0.1, 0.1],
+                    ],
+                ]
+            ]
+        )
+        .to(device)
+        .requires_grad_()
+    )
+    targets = torch.Tensor([[1, 2]]).to(device).int()
+    probs_length = torch.Tensor([2.0]).to(device).int()
```
Comment (carolineechen) [inline tests/unittests/test_losses.py:171]: note that SpeechBrain's unittest has the value `1.0` here instead. it looks like they input `T`, `U` values for computing the cuda kernel rather than `maxT`, `maxU`, which these tensors correspond to in torchaudio's implementation and which are computed in the loss function in SpeechBrain

File: tests/unittests/test_losses.py [hunk 2]
```diff
@@ -137,3 +138,43 @@ def test_transducer_loss():
...
+                    [
+                        [0.1, 0.6, 0.1, 0.1, 0.1],
+                        [0.1, 0.1, 0.2, 0.1, 0.1],
+                        [0.7, 0.1, 0.2, 0.1, 0.1],
+                    ],
+                ]
+            ]
+        )
+        .to(device)
+        .requires_grad_()
```
Comment (carolineechen) [inline tests/unittests/test_losses.py:168]: also removed `log_softmax` here (using `fused_log_softmax=True` default), but should not affect the loss

File: tests/unittests/test_losses.py [hunk 3]
```diff
@@ -137,3 +138,43 @@ def test_transducer_loss():
...
+    probs_length = torch.Tensor([2.0]).to(device).int()
+    target_length = torch.Tensor([2.0]).to(device).int()
+
+    out_cost = rnnt_loss(
+        log_probs, targets, probs_length, target_length, blank=0,
+        reuse_logits_for_grads=False
+    )
+    loss = out_cost.mean()
+    loss.backward()    
+    assert loss.item() == 2.247833251953125
```
Comment (carolineechen) [inline tests/unittests/test_losses.py:180]: Running this [code snippet](https://github.com/HawkAaron/warp-transducer/tree/master/pytorch_binding) from warp-transducer and printing the output produces the same loss value as torchaudio here, `4.49566650390625`. Using probs_length/target_length as 1 instead of 2 using warp-transducer also produces an error.

Comment (carolineechen) [reply tests/unittests/test_losses.py:180]: Speech Brain's loss includes [time-based normalization](https://github.com/speechbrain/speechbrain/blob/develop/speechbrain/nnet/loss/transducer_loss.py#L103). Removing the 2 instances of normalization in their code results in the same loss as torchaudio and warp-transducer
